### PR TITLE
Fixed java -> scala type conversions in CompiledExecutionResult

### PR DESCRIPTION
**Pull request by lutovich**

### CODE REVIEW

**Comment by lutovich** [comment]
@pontusmelke could you please review this PR

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala [hunk 1]
```diff
@@ -216,9 +216,15 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
...
-    case iter: Map[_, _] => Eagerly.immutableMapValues(iter, makeValueJavaCompatible).asJava
-    case x               => x
+    case seq: Seq[_] => seq.map(makeValueJavaCompatible).asJava
+    case map: Map[_, _] => Eagerly.immutableMapValues(map, makeValueJavaCompatible).asJava
+    case x => x
+  }
+
+  private def makeValueScalaCompatible(value: Any): Any = value match {
+    case list: java.util.List[_] => list.asScala.map(makeValueScalaCompatible).toList
+    case map: java.util.Map[_, _] => map.asScala.map { case (k, v) => k -> makeValueScalaCompatible(v) }.toMap
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala]: I think you want to do `Eagerly.immutableMapValues(map.asScala, makeValueScalaCompatible)` Not sure if it matters in this case but it is at least constistent

**Comment by pontusmelke** [comment]
I think we should remove `makeJavaCompatible` from https://github.com/lutovich/neo4j/blob/2.3-fix-types-in-execution-result/community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala#L64

**Comment by lutovich** [comment]
@pontusmelke I do not think it is possible now. We get result as java collection then convert it to scala in `ensureIterator()`. That's why `javaIterator()` needs to do one more conversion.

**Comment by pontusmelke** [comment]
@lutovich I guess that't ok but it feels a bit backward: We produce result as java collections in `accept` and then when we turn it into an iterator, we create scala collections and then when creating the javaIterator we turn it back into java again. The java stuff is the important path since that is what is returned in the actual `Result` when runing `db.execute(…)` so it would be nicer if the java-results goes straight through and do the mapping and translating for scala results only.

I think we need to change the `innerIterator` to be a java iterator over java maps and then do the translation whenever we ask for scala stuff. But maybe we can wait with that since I think we should turn `CompileExecutionResult` into a java class altogether.

**Comment by lutovich** [comment]
@pontusmelke changed code so that we store results as java iterator. Does it look better now?

**Pull request merged**